package/src/plugins/add-virtual-imports.ts

```diff
@@ -0,0 +1,20 @@
+import { definePlugin } from "../core/define-plugin.js";
+import { addVirtualImports } from "../utilities/add-virtual-imports.js";
+
+const pluginIndexMap: Record<string, number> = {};
+
+export const addVirtualImportsPlugin = definePlugin({
+	name: "addVirtualImports",
+	hook: "astro:config:setup",
+	implementation:
+		({ updateConfig }, { name }) =>
+		(imports: Record<string, string>) => {
+			pluginIndexMap[name] ??= -1;
+			pluginIndexMap[name]++;
+			addVirtualImports({
+				name: `${name}-${pluginIndexMap[name]}`,
+				imports,
+				updateConfig,
+			});
+		},
+});
```

Comment:
I don't think an external map is needed to keep track of the index, the state can be contained in a closure instead of a global.

```suggestion
export const addVirtualImportsPlugin = definePlugin({
	name: "addVirtualImports",
	hook: "astro:config:setup",
	implementation:
		({ updateConfig }, { name }) => {
		  let counter = 1;
		  (imports: Record<string, string>) => {
			  addVirtualImports({
				  name: `${name}-${counter++}`,
				  imports,
				  updateConfig,
			  });
		  },
		},
});
```

Reply:
Sorry for the mix of tabs and spaces. Gotta format locally if you choose to apply from GH web

Reply:
🤦‍♂️ Ya that makes a lot more sense! Ill fix formatting locally so you get co-author credit